src/http_server.h

```diff
@@ -230,6 +253,91 @@ class HTTPAPIServer : public HTTPServer {
     std::atomic<uint32_t> response_count_;
   };
 
+  class GenerateRequestClass : public InferRequestClass {
```

Comment:
nit: Can we move this to  a separate file? Something similar to how we broke down composing handlers in gRPC?
Makes sense to just start with Generate handler.

Reply:
That is planned, but not in this PR as I would like to move out InferRequestClass as well